dataOverlay.js

```diff
@@ -0,0 +1,116 @@
+const fs = require('fs');
+const path = require('path');
+
+const DATA_TYPE_COMPONENTS = {
+  transcriptomics: 'gene',
+  metabolomics: 'compartmentalizedMetabolite',
+};
+
+/*
+ * This function transforms data files into files that are ready
+ * to be used in the Metabolic Atlas website.
+ * Example:
+ * `modelDir`: ../data-files/integrated-models/Human-GEM
+ * output index file: ./dataOverlay/Human-GEM/index.json
+ * output data source file: ./dataOverlay/Human-GEM/transcriptomics/protein1.mock.tsv
+ */
+const processDataOverlayFiles = ({ modelDir, outDir, componentIdDict }) => {
+  const filesDir = `${modelDir}/dataOverlay`;
+  if (!fs.existsSync(filesDir)) {
+    return;
+  }
+
+  const modelOutDir = getModelOutDir({ modelDir, outDir });
+
+  const dataOverlayFiles = {};
+  const dataTypes = fs
+    .readdirSync(filesDir, { withFileTypes: true })
+    .filter((dirent) => dirent.isDirectory())
+    .map((dirent) => dirent.name);
+
+  const dataSourcesDict = dataTypes.reduce(
+    (obj, dt) => ({
+      ...obj,
+      [dt]: parseIndexFile(
+        fs.readFileSync(`${filesDir}/${dt}/index.tsv`, 'utf8'),
+      ),
+    }),
+    {},
+  );
+
+  fs.writeFileSync(
+    `${modelOutDir}/index.json`,
+    JSON.stringify(dataSourcesDict),
+    'utf8',
+  );
+
+  for (const [dt, metadataList] of Object.entries(dataSourcesDict)) {
+    const componentType = DATA_TYPE_COMPONENTS[dt];
+    const componentIdSet = new Set(Object.keys(componentIdDict[componentType]));
+
+    for (const { filename } of metadataList) {
+      const inputFile = fs.readFileSync(
+        `${filesDir}/${dt}/${filename}`,
+        'utf8',
+      );
+      const condensedFile = condenseDataSourceFile({
+        inputFile,
+        componentIdSet,
+      });
+
+      const dataSourceOutDir = `${modelOutDir}/${dt}`;
+      if (!fs.existsSync(`${dataSourceOutDir}`)) {
+        fs.mkdirSync(`${dataSourceOutDir}`);
+      }
+
+      fs.writeFileSync(
+        `${dataSourceOutDir}/${filename}`,
+        condensedFile,
```

Comment:
I kind of prefer to add a newline at the end of output file (the same as the original files in the folder `data-files`, but it is up to you to add it or not :)

Reply:
Ah I see, if we have a standard I should probably follow it. In general I like to add a newline at the end for code source files as well, but I feel like for this type of data files it's more convenient for parsers if there is no such newline.

Reply:
make sense.